libcyphal_demo/src/application.hpp

```diff
@@ -19,6 +23,81 @@
 class Application final
 {
 public:
+    struct Regs
+    {
+        /// Defines general purpose string parameter exposed as mutable register.
+        ///
+        template <std::size_t N>
+        struct StringParam
+        {
+            StringParam(const libcyphal::application::registry::IRegister::Name name,
+                        libcyphal::application::registry::Registry&             registry,
+                        const cetl::string_view                                 initial_value)
+                : value_{initial_value}
+                , memory_{registry.memory()}
+                , register_{registry.route(
+                      name,
+                      [this] {
+                          //
+                          return makeStringValue(value_);
+                      },
+                      [this](const auto& value) -> cetl::optional<libcyphal::application::registry::SetError> {
+                          //
+                          if (value.is_string())
+                          {
+                              this->value_ = libcyphal::application::registry::makeStringView(value.get_string().value);
+                              return cetl::nullopt;
+                          }
+                          return libcyphal::application::registry::SetError::Semantics;
+                      })}
+            {
+            }
+
+            CETL_NODISCARD platform::String<N>& value()
+            {
+                return value_;
+            }
+
+        private:
+            CETL_NODISCARD libcyphal::application::registry::IRegister::Value makeStringValue(
+                const cetl::string_view sv) const
+            {
+                using Value = libcyphal::application::registry::IRegister::Value;
+
+                const Value::allocator_type allocator{&memory_};
+                Value                       value{allocator};
+                auto&                       str = value.set_string();
+                std::copy(sv.begin(), sv.end(), std::back_inserter(str.value));
+                return value;
+            }
+
+            platform::String<N>                                            value_;
+            cetl::pmr::memory_resource&                                    memory_;
+            libcyphal::application::registry::Register<sizeof(void*) * 12> register_;
```

Comment:
Based on feedback in the Matrix Cyphal channel, would probably be good to document the use of 12 as a magic number, something like:

```
///
/// Size of the Register is arbitrary (12 pointers)
///
```

Reply:
addressed in the next pr